### PR TITLE
Bluespace Light Replacer!!

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
port of https://github.com/BeeStation/BeeStation-Hornet/pull/7414

Adds a bluespace light replacer, which acts as a researchable upgrade for the light replacer. You can replace lights without having to be next to them, through bluespace tech! Due to its unstable nature (and game balance), emagging it is impossible. Also standard light replacer is moved to a roundstart

## Why It's Good For The Game
Lets janitors/others replace lights in restricted areas as long as theres a window line of sight, which helps janitors with not having to beg crew/ai to let them in and the like

## Changelog

:cl:
add: bluespace light replacer
/:cl:

### CODE REVIEW

**Comment by Jackal-boop** [comment]
There is already a blue space light replacer that can be gained through BEPIS tech. Why do we need 2?

**Comment by etherware-novice** [comment]
wait there is?

**Comment by JohnFulpWillard** [comment]
I think it's ``/obj/item/construction/rld``

**Comment by Jackal-boop** [comment]
code/game/objects/items/devices/lightreplacer.dm

**Comment by Jackal-boop** [comment]
I do think this should be taken out of bluespace tech though. Its super useful and BEPIS tech is far too rare. Moving it to advanced engineering or janitorial tech would be awesome

**Comment by Jackal-boop** [comment]
> code/game/objects/items/devices/lightreplacer.dm

Wait this is the wrong thing. Where the hell is the rapid replacer

**Comment by BlueMemesauce** [comment]
> I think it's `/obj/item/construction/rld`

That build new lights and doesn't replace them. also it can't be used at range

**Comment by etherware-novice** [comment]
so thats a completely different thing to a light replacer upgrade, im good then

**Comment by Baskett55** [comment]
Isn't that bepis item under cargo goodies?

**Comment by Jackal-boop** [comment]
> Isn't that bepis item under cargo goodies?

Yeah. It should probably be placed into a Janitor mail gift or a service/engineering crate

**Comment by etherware-novice** [comment]
alright, but does this conflict with my bluespace light replacer?

**Review by ShizCalev — APPROVED** [review]

**Comment by ShizCalev** [comment]
Code is good. Will merge once spriters ok it.

**Review by Twaticus — APPROVED** [review]

**Review by Imaginos16 — APPROVED** [review]
Passable.

**Comment by etherware-novice** [comment]
oh come on

File: code/game/objects/items/devices/lightreplacer.dm [hunk 1]
```diff
@@ -246,3 +252,16 @@
...
+	desc = "A modified light replacer that zaps lights into place. Refill with broken or working lightbulbs, or sheets of glass."
+	icon_state = "lightreplacer_blue0"
+	bluespace_toggle = TRUE
+
+/obj/item/lightreplacer/blue/emag_act()
+	return  // balancing against longrange explosions
+
+/obj/item/lightreplacer/blue/update_icon(updates)
+	. = ..()
+	icon_state = initial(icon_state)
```
[the remark this answers is not in the document]
Comment (Fikou) [reply code/game/objects/items/devices/lightreplacer.dm:267]: whats the point

[the remark this answers is not in the document]
Comment (etherware-novice) [reply code/game/objects/items/devices/lightreplacer.dm:267]: good point actually

**Pull request merged**